### PR TITLE
Travis rework

### PR DESCRIPTION
**Pull request by whot**
Same as in libwacom, mostly copy-paste from there anyway.

### CODE REVIEW

**Comment by whot** [comment]
Well, that was easier than expected. tests pass and the travis log looks correct, so unless there's an issue with the coverity bits, this should be it.

Note that we get a few extra things now too, srcdir != builddir, distcheck, etc.

File: .travis.yml [hunk 1]
```diff
@@ -15,29 +15,47 @@ addons:
...
-script:
-  - if [[ x"$TRAVIS_EVENT_TYPE" = "xcron" ]]; then export DO_COVERITY="YES"; fi
-  - if [[ x"$BUILDTYPE" = "xcoverity" && x"$DO_COVERITY" = "xYES" ]]; then bash ./coverity.sh; fi
-  - if [[ x"$BUILDTYPE" != "xcoverity" ]]; then make CFLAGS="$CFLAGS" && make check; fi
+jobs:
+  include:
+    - stage: coverity
+      compiler: 'gcc'
+      env:
+        - BUILDTYPE="coverity"
```
Comment (jigpu) [inline .travis.yml]: Just noticed that this variable was also left in libwacom. It shouldn't be necessary now that Coverity is done as its own job.

File: .travis.yml [hunk 2]
```diff
@@ -15,29 +15,47 @@ addons:
...
+        - COVERITY_SCAN_PROJECT_NAME="linuxwacom/libwacom"
```
Comment (jigpu) [inline .travis.yml]: xf86-input-wacom :)

File: .travis.yml [hunk 3]
```diff
@@ -15,29 +15,47 @@ addons:
...
+        - COVERITY_SCAN_NOTIFICATION_EMAIL="killertofu@gmail.com"
+        - COVERITY_SCAN_BUILD_COMMAND_PREPEND=""
+        - COVERITY_SCAN_BUILD_COMMAND="make"
+        - COVERITY_SCAN_BRANCH_PATTERN=".*"
+        - COVERITY_URL="https://scan.coverity.com/scripts/travisci_build_coverity_scan.sh"
+        - secure: "o99zVPgb3kU80P6nC8gQWIr4UdRa7+d+K8g8Gclmzgy980NSk1MBLtD1oqoXThAx4BGnW6XDIFG8RRsl0QkE7gsZyOw5WwcRWrsK2ZdCMNngLxhG7QKF/l9y8WMmUaqSXW9WTgyheqIH5QuEYtkoGqjQBP3fNwPDpGT720H3jSxibRjkwUUyo5Shy5LS9I3d6uL1/7nwgcdAAocYCl5PWm/swVXwrFBvFENKCp8i/0LDXmJ9g/NrLqKljiMXoWxP8SUsTiUk+D3Q4s0/Izy7SbZxbQMoG+Z2Et6iu0r7boxqw2McWt6WqovoAFhm4ZSNmBYbRhNFxMCbabrBNWcr0k58BvnpDA/5TAbeesudY4Tb4tebq8oQJUcQqYgxI3YOnaQQTSjGWHdQLoYIJnGu2ctp7vP8pnxaEmPxoeXMojEXiIZe2426Ysf8DfQ9cKxXYKEwANHWu6Mwo7Kf9KZ0fgeSTU4KKgTIDuDpP7LIquWSuDp7lA9/zGqyjdHslps3VDFRPXkUUizqRE/clyZm4bmkWF5QFzbuuKDSSi+roGFEvBuL5SGOv13uq4stEo9k4Tkkv9dAP7nofSo5deWBHXcwaYlGMWzQKOykanxq/JIEUCPp+Cmfglee6kjRjeNwKwrrxg1nDDLarp/3B1Qnvua4CVrjPav39yHAKNFPtXg="
```
Comment (jigpu) [inline .travis.yml]: This secure variable is giving Travis heartburn for some yet-to-be-determined reason. I'll be following-up on this...

**Comment by whot** [comment]
both of the easy fixes fixed with a force-push

File: .travis.yml (hunk 3 above)
Comment (jigpu) [reply .travis.yml]: Nevermind. Just a case of PEBKAC.

**Pull request merged**